tests/evaluate/test_default_evaluator.py

```diff
@@ -1986,12 +1986,52 @@ def language_model(inputs: list[str]) -> list[str]:
     return inputs
 
 
+def validate_question_answering_logged_data(logged_data, with_targets=True):
+    if with_targets:
+        columns = [
+            "question",
+            "answer",
+            "outputs",
+            "toxicity",
+            "flesch_kincaid_grade_level",
+            "ari_grade_level",
+            "perplexity",
+        ]
+    else:
+        columns = [
+            "question",
+            "outputs",
+            "toxicity",
+            "flesch_kincaid_grade_level",
+            "ari_grade_level",
+            "perplexity",
+        ]
+
+    assert logged_data.columns.tolist() == columns
+
+    assert logged_data["question"].tolist() == ["words random", "This is a sentence."]
+    assert logged_data["outputs"].tolist() == ["words random", "This is a sentence."]
+    assert logged_data["toxicity"][0] < 0.5
+    assert logged_data["toxicity"][1] < 0.5
+    assert logged_data["perplexity"][0] > logged_data["perplexity"][1]
+    for grade in logged_data["flesch_kincaid_grade_level"] + logged_data["ari_grade_level"]:
+        assert isinstance(grade, (int, float))
```

Comment:
Do we need `int`?

Reply:
We don't necessarily need it here but I think we wanted to make it as general as possible in the case that we change the model. Could change it to float though to be more clear about what the model is returning.